### ISSUE TITLE
compiliation fails after last update

### ISSUE THREAD

**Issue by dreith** [msg-byu/enumlib#12]
Hello everybody, 

I can't seem to be able to compile enumlib after the last update using either gfortran or ifort. 
The compilation error is:

```
gfortran -O3 -ffree-line-length-none -fno-underscoring -I../../symlib/src -c tree_class.f90
tree_class.f90:36.24:

     type(GroupList)  :: G
                        1
Error: Derived type at (1) has not been previously defined and so cannot appear in a derived type definition
tree_class.f90:106.43:

          rlabeling = labeling(self%G%layer(self%depth())%perms(ig,:))
                                           1
Error: Derived type 'grouplist' at (1) is being used before it is defined
tree_class.f90:111.25:

             self%G%layer(self%depth()+1)%perms(cg,:) = self%G%layer(self%depth
                         1
Error: Derived type 'grouplist' at (1) is being used before it is defined
tree_class.f90:173.25:

    allocate(self%G%layer(self%k)) !Allocate number of perm
                         1
Error: Derived type 'grouplist' at (1) is being used before it is defined
tree_class.f90:176.25:

    allocate(self%G%layer(1)%perms(size(generators,1),size(generators,2)))
                         1
Error: Derived type 'grouplist' at (1) is being used before it is defined
tree_class.f90:177.16:

    self%G%layer(1)%perms = generators
                1
Error: Derived type 'grouplist' at (1) is being used before it is defined
tree_class.f90:181.35:

          call grouper(self%G%layer(1)%perms)
                                   1
Error: Derived type 'grouplist' at (1) is being used before it is defined
tree_class.f90:183.31:
```

**Comment by wsmorgan** [comment]
When we updated the repo we moved GroupList and permList from tree_class.f90 to enumeration_types.f90 but apparently they weren't declared as public at that point as well. This issue will be fixed in Revision 0.0.2.

**Comment by dreith** [comment]
works in the new revision
